lib/pbio/sys/status.c

```diff
@@ -35,6 +37,30 @@ static void pbsys_status_update_flag(pbio_pybricks_status_t status, bool set) {
         (process_data_t)status);
 }
 
+/**
+ * Writes Pybricks status report command to @p buf
+ *
+ * @param [in]  buf        The buffer to hold the binary data.
+ * @return                 The number of bytes written to @p buf.
+ */
+uint32_t pbsys_status_write_status_report(uint8_t *buf) {
```

Comment:
This looks out of place. Would expect it to be with the rest of the similar functions in `pybricks.c`.

We can just add a `pbsys_status_get_program_id()` instead.

Reply:
I considered this initially, but assembling the `status` payload in `status.c` (where the status state is kept) does seem about right.

We could add a pure function to `protocol.c` to be complete I suppose. This was initially omitted it since it almost repeats the other one.

Both would seem a bit more natural than the current approach, where the Bluetooth system driver is passing its version of the system state to build the status message.